src/views/forms/FormSubAccountCreation/FormSubAccountCreationTs.ts

```diff
@@ -233,25 +248,82 @@ export class FormSubAccountCreationTs extends Vue {
+          this.$store.dispatch('notification/ADD_SUCCESS', NotificationType.OPERATION_SUCCESS)
+          this.$emit('submit', this.formItems)
+        })
+        .catch((err) => console.log(err))
+    } else {
+      // - get next path
+      const nextPath = this.paths.getNextAccountPath(this.knownPaths)
+
+      this.$store.dispatch('diagnostic/ADD_DEBUG', 'Adding child account with derivation path: ' + nextPath)
+
+      // - decrypt mnemonic
+      const encSeed = this.currentProfile.seed
+      const passphrase = Crypto.decrypt(encSeed, this.currentPassword.value)
+      const mnemonic = new MnemonicPassPhrase(passphrase)
+
+      // create account by mnemonic
+      return this.accountService.getChildAccountByPath(
+        this.currentProfile,
+        this.currentPassword,
+        mnemonic,
+        nextPath,
+        this.networkType,
+        childAccountName,
+      )
+    }
+  }
 
-    // - get next path
-    const nextPath = this.paths.getNextAccountPath(this.knownPaths)
-
-    this.$store.dispatch('diagnostic/ADD_DEBUG', 'Adding child account with derivation path: ' + nextPath)
-
-    // - decrypt mnemonic
-    const encSeed = this.currentProfile.seed
-    const passphrase = Crypto.decrypt(encSeed, this.currentPassword.value)
-    const mnemonic = new MnemonicPassPhrase(passphrase)
-
-    // create account by mnemonic
-    return this.accountService.getChildAccountByPath(
-      this.currentProfile,
-      this.currentPassword,
-      mnemonic,
-      nextPath,
-      this.networkType,
-      childAccountName,
-    )
+  async importSubAccountFromLedger(childAccountName: string): Promise<AccountModel> | null {
+    const subAccountName = childAccountName
+    const accountPath = this.currentAccount.path
+    const currentAccountIndex = accountPath.substring(accountPath.length - 2, accountPath.length - 1)
+    const numAccount = this.knownPaths.length
+    let accountIndex
+    if (numAccount <= Number(currentAccountIndex)) {
+      accountIndex = numAccount + Number(currentAccountIndex)
+    } else {
+      accountIndex = numAccount + 1
+    }
+    try {
+      this.$Notice.success({
+        title: this['$t']('Verify information in your device!') + '',
+      })
+      const transport = await TransportWebUSB.create()
+      const symbolLedger = new SymbolLedger(transport, 'XYM')
+      const accountResult = await symbolLedger.getAccount(`m/44'/4343'/${this.networkType}'/0'/${accountIndex}'`)
+      const { address, publicKey, path } = accountResult
+      transport.close()
+      const accName = Object.values(this.currentAccount)[1]
+      return {
+        id: SimpleObjectStorage.generateIdentifier(),
+        name: subAccountName,
+        profileName: accName,
+        node: '',
+        type: AccountType.fromDescriptor('Seed'),
+        address: address.toUpperCase(),
+        publicKey: publicKey.toUpperCase(),
+        encryptedPrivateKey: '',
+        path: path,
+        isMultisig: false,
+      }
```

Comment:
Shouldn't this return a Wallet model ?

Reply:
As I know, currently in the master branch, you don't use Wallet Model. I change this to suitable for the change in master.

Reply:
@fboucquez ?

Reply:
WalletModel has been replaced by AccountModel. it's a yellow flag if you are creating an object this big without type.

Reply:
I used the same structure as the others functions in AccountService.ts.